### PR TITLE
nmap: migrate to gcc11

### PR DESCRIPTION
**Pull request by davidkaroly**
similar to #7822

### CODE REVIEW

File: net-analyzer/nmap/nmap-3.81.recipe [hunk 1]
```diff
@@ -56,8 +62,9 @@ BUILD()
 	libtoolize --force --copy --install
 	popd
 
-	runConfigure --omit-dirs "dataRootDir docDir" \
-		./configure  --without-nmapfe LDFLAGS=-lnetwork
+	runConfigure --omit-dirs "binDir dataRootDir docDir" \
+		./configure --bindir="$prefix/bin" \
```
Comment (Begasus) [inline net-analyzer/nmap/nmap-3.81.recipe:66]: Same here for $commandBinDir: https://github.com/Begasus/haikuports/wiki/Useful-variables-for-recipes#commandbindir-and-commandsuffix

Comment (davidkaroly) [reply net-analyzer/nmap/nmap-3.81.recipe:66]: even better there should be a similar helper for CMake as well (which needs some rework anyway as currently `cmakeDirArgs` are all absolute which does not work so well on some recipes)

Comment (Begasus) [reply net-analyzer/nmap/nmap-3.81.recipe:66]: This should be dealth with at haikuporter I guess, @davidkaroly good to go?

Comment (davidkaroly) [reply net-analyzer/nmap/nmap-3.81.recipe:66]: i think so

**Comment by Begasus** [comment]
Build failure on both buildmaster :/
https://build.haiku-os.org/buildmaster/master/x86_64/logviewer.html?buildruns/9312/builds/80379.log

**Comment by davidkaroly** [comment]
Strange that it failed on the x86_64 as well, it should have been only a rebuild for 64bit without any change.

**Comment by pulkomandy** [comment]
The error is (for future reference, I think the buildmaster won't keep the log forever):

{{{
g++ -lnetwork -Lnbase -Lnsock/src/  -o nmap main.o nmap.o targets.o tcpip.o nmap_error.o utils.o idle_scan.o osscan.o output.o scan_engine.o timing.o charpool.o services.o protocols.o nmap_rpc.o portlist.o NmapOps.o TargetGroup.o Target.o FingerPrintResults.o service_scan.o NmapOutputTable.o MACLookup.o  -lnbase -lnsock -lpcre -lpcap -lssl -lcrypto -lm
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/13.2.0/../../../../x86_64-unknown-haiku/bin/ld: output.o:/boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/13.2.0/include/c++/x86_64-unknown-haiku/bits/gthr-default.h:250:(.data.rel.local+0x0): undefined reference to `__gthrw_pthread_cancel(_pthread_thread*)'
}}}

This function is supposed to be a weak alias to pthread_cancel defined in libgcc (buildtools/gcc/libgcc/gthr-posix.h).

That file is included in the libgcc build from gcc/libgcc/configure with a snippet generated from gcc/config/gthr.m4 which is invoked from libgcc/configure.ac. The parameter is target_thread_file which is initialized from the "Thread model:" value printed by gcc -v.

There are also some defines that decide wether the function is enabled or not in gthread-posix.h, mainly depending on SUPPORTS_WEAK and GTHREAD_USE_WEAK. Could it be that one of those changed values at some point between building libgcc and compiling nmap? 

Earlier in the log we see:

```
In file included from /boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/13.2.0/include/c++/x86_64-unknown-haiku/bits/gthr.h:148,
                from /boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/13.2.0/include/c++/ext/atomicity.h:35,
                from /boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/13.2.0/include/c++/bits/cow_string.h:37,
                from /boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/13.2.0/include/c++/bits/basic_string.h:51,
                from /boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/13.2.0/include/c++/string:54,
                from output.cc:112:
/boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/13.2.0/include/c++/x86_64-unknown-haiku/bits/gthr-default.h: At global scope:
/boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/13.2.0/include/c++/x86_64-unknown-haiku/bits/gthr-default.h:112:1: warning: 'int __gthrw_pthread_cancel(pthread_t)' used but never defined
 112 | __gthrw(pthread_cancel)
```

Here we are in libstdc++ (notice the "c++" in the include path) and it is similar to ghtread_posix, but it uses different defines to control wether the __gthr_ weak alias are generated: __GXX_WEAK__ and _GLIBCXX_GTHREAD_USE_WEAK. Do these have the same value than the corresponding ones in libgcc?

**Pull request merged**